packages/@aws-cdk/aws-logs/lib/log-retention.ts

```diff
@@ -145,7 +167,7 @@ class LogRetentionFunction extends Construct implements cdk.ITaggable {
     });
     // Duplicate statements will be deduplicated by `PolicyDocument`
     role.addToPrincipalPolicy(new iam.PolicyStatement({
-      actions: ['logs:PutRetentionPolicy', 'logs:DeleteRetentionPolicy'],
+      actions: ['logs:PutRetentionPolicy', 'logs:DeleteRetentionPolicy', 'logs:DeleteLogGroup'],
```

Comment:
I think this is dangerous. We need to scope this role down to only be allowed to delete the log group that this is applied to.

Reply:
Thank you! I can only find the action 'logs:DeleteLogGroup' in the list. [permissions-reference](https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/permissions-reference-cwl.html). Is there other way to limit permission of deleting log groups?

Reply:
Deletion of the log group is fine but we need to scope down the amount of resources this can act on. It should be limited to the log group in question, not be a `*` resource. Note that you'll also need permissions to delete the log streams within the log group.